herald.py

```diff
@@ -129,26 +131,41 @@ async def runHerald(ctx, args):
         return
 
     pickle.dump(heraldDict, open("herald/heraldUsers.p", "wb"))
+    return
 
 async def playHerald(member):
 
     # Only run if user has selected audio
     if member.id in heraldDict.keys():
 
-        print(f'{member.name} has played Herald!')
+        current_time = datetime.datetime.now()
+
+        #only run if user has never joined voice channel or the time between joining exceeds delay
+        if (isinstance(heraldDict[member.id], type(None))) or ((current_time - heraldDict[member.id].lastUseTime).total_seconds() > wait_timer):
+
+            print(f'{member.name} has played Herald!')
+
+            #set current time to use for future vc joins
+            heraldDict[member.id].lastUseTime = current_time
 
-        # Connect to voice channel
-        time.sleep(0.5)
-        vc = await member.voice.channel.connect()
+            # Connect to voice channel
+            time.sleep(0.5)
+            vc = await member.voice.channel.connect()
 
-        # Set and start audio
-        audio = discord.FFmpegPCMAudio(heraldDict[member.id].editedMp3Link)
-        time.sleep(0.5)
-        vc.play(discord.PCMVolumeTransformer(audio, volume=0.7))
+            # Set and start audio
+            audio = discord.FFmpegPCMAudio(heraldDict[member.id].editedMp3Link)
+            time.sleep(0.5)
+            vc.play(discord.PCMVolumeTransformer(audio, volume=0.7))
 
-        # Wait for duration
-        time.sleep(heraldDict[member.id].duration)
+            # Wait for duration
+            time.sleep(heraldDict[member.id].duration)
 
-        # Stop and disconnect when done
-        vc.stop()
-        await vc.disconnect()
\ No newline at end of file
+            # Stop and disconnect when done
+            vc.stop()
+            await vc.disconnect()
+
+        else:
```

Comment:
Do we need an else here since nothing is happening after (if we move the pickle dump to inside the if)

Reply:
We could remove the else! I put that there in case you wanted the bot to say something or print to the terminal.